### PR TITLE
Add shell.nix

### PR DESCRIPTION
**Pull request by mlieberman85**
Simple shell.nix for anyone developing against guac who uses nix
or nixos.

Signed-off-by: Michael Lieberman <mlieberman85@gmail.com>

### CODE REVIEW

File: shell.nix [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{ pkgs ? import <nixpkgs> {} }:
+with pkgs;
+mkShell {
+  buildInputs = [
+    gcc
+    go_1_18
+    gopls
+    gotests
```
Comment (lumjjb) [inline shell.nix:8]: would this need the linting tool for golangci-lint and goimports for the make checks?

**Review by lumjjb — APPROVED** [review]
minor comment, lgtm

File: shell.nix (hunk 1 above)
Comment (mlieberman85) [reply shell.nix:8]: Good point let me check

**Comment by mlieberman85** [comment]
I will add the other things once the makefile is merged

**Pull request merged**